schemas/graph.schema.js

```diff
@@ -2,9 +2,59 @@ const mongoose = require("mongoose");
 const Schema = mongoose.Schema;
 const { DEFAULT_USER } = require("../constants");
 
+const Vertex = new Schema({
+  id: {
+    type: String,
+    required: true,
+  },
+  location: {
+    x: {
+      type: Number,
+      required: true,
+    },
+    y: {
+      type: Number,
+      required: true,
+    },
+  },
+  color: {
+    type: String,
+    default: "fca311",
+  },
+  distance: {
+    type: Number,
+    default: Infinity,
+  },
+  PI: {
+    type: Schema.Types.ObjectId,
```

Comment:
What does `PI` means ?

Reply:
in algorithm related to graphs, the PI is the common way to address to the previous vertex from whence the algorithm arrived at that vertex.(PI as in the symbol of 3.14....)